### DISCUSSION TITLE
How to simulate super-gauss laser?

### DISCUSSION THREAD

**Discussion by Hatisa** [epochpic/epoch#758] [Q&A]
I want to simulate a continuous super-gauss laser. My input settings are as follows.
![image](https://github.com/user-attachments/assets/4fd26f83-0c86-4916-943e-4666ab007bd2)
I just use supergauss(y, 0, 6 * micron, 10) to simulate the laser.(2D PIC)
![image](https://github.com/user-attachments/assets/a9cff795-3bb5-45d5-bdd9-da9352a7e2e4)
However, to simulate the laser, we also need to know the phase. Gauss laser phase are as follows:
![image](https://github.com/user-attachments/assets/461f0d88-d7cb-41c0-aaca-6409510d0207)
And Super-Gauss Laser phase:
![image](https://github.com/user-attachments/assets/3c180c10-de60-4290-88bd-f8375f807954)
and
![image](https://github.com/user-attachments/assets/d9a42648-ed9f-4bdb-90a2-ba1d8536cc76)
I'm having a hard time programming the phase shift described above. 
I'm seeking guidance from experts on whether my simulation approach is sound and how to simulate super-Gaussian beams effectively.
Thanks!

**Comment by Status-Mirror** [comment]
Hey @Hatisa,

You may have already come across our [Gaussian beam](https://epochpic.github.io/quickstart/basic_examples/laser.html#focussing_a_gaussian_beam) demo. To achieve this, we take the equation you label "Gauss laser phase", and evaluate all the $z$ terms at the simulation boundary. You can probably do something similar with your $\phi(r_2,z)$.

I'm less familiar with your particular profile, but I think EPOCH might struggle to evaluate some of your terms in the input deck. Instead, you might find it better to calculate the numerical values of $\phi(r,z)$ at the simulation boundary in a separate program, to obtain a table of $r$ and $\phi$ value pairs. Then, you can set the laser terms in a similar way to how they're done in the [Gaussian beam](https://epochpic.github.io/quickstart/basic_examples/laser.html#focussing_a_gaussian_beam)  demo using the `interpolate` function in the [maths parser](https://epochpic.github.io/documentation/code_details/maths_parser.html).

The net effect is that for each $r_2$, (or $y$ in EPOCH), the cell position will be substituted into your `interpolate` function, and the phase will be returned. Each cell will know what the phase should be, and you can hopefully use this to inject your pulse.

Hope this helps,
Stuart
